packages/workspace/src/browser/workspace-service.ts

```diff
@@ -77,10 +79,25 @@ export class WorkspaceService implements FrontendApplicationContribution {
      */
     onStop(app: FrontendApplication): void {
         if (this._root) {
-            this.server.setRoot(this._root.uri);
+            this.server.setWorkspace(this._root.uri);
+        }
+    }
+
+    async onStart() {
+        const allWorkspace = await this.recentWorkspaces();
+        if (allWorkspace.length > 0) {
+            this.hasWorkspace = true;
         }
     }
 
+    get hasHistory(): boolean {
+        return this.hasWorkspace;
+    }
+
+    async recentWorkspaces(): Promise<string[]> {
+        return this.server.getRecentWorkspaces();
```

Comment:
Could it happen that `recentWorkspaces` gets modified in the meantime and `this.hasWorkspace` is lying? Should not we cache `recentWorkspaces` in `onStart` and return them here always?

Reply:
@akosyakov It is ok that recentWorkspace get modified. With on start, we verify there is  at least one workspace to allow the sensitivity of the button.
Scenario:
- never opened -> nothing in recentWorkspaces  or files does not exist -> return false
- open a workspace once -> return true
- open once or more workspaces, close all -> return true even if all workspaces are closed.
The only time we can see the button is sensitive with no workspace would be: start Client, open a workspace, then close that workspace. Keep the client open and you manually delete the file "recentWorkspace.json". The button sensitivity would still allow to select, but the list returned would be empty, nothing to select.